### PR TITLE
[IMP] Port TMS report to new API

### PR DESCRIPTION
**Pull request by seb-elico**
Port TMS report to new API

### CODE REVIEW

**Comment by seb-elico** [comment]
@victormartinelicocorp @Wizr @Zhouhao12345 can you answer the `FIXME`s in my code please?

File: timesheet_activity_report/__openerp__.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 {
     'name': 'Timesheet Activities Report',
-    'version': '8.0.1.0.3',
+    'version': '8.0.1.1.0', #FIXME is it the right version? I assume the module needs to be upgraded...
```
Comment (elicoidal) [inline timesheet_activity_report/__openerp__.py]: correct: restart+upgrade, no migration (https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers)

**Review by elicoidal — APPROVED** [review]
LGTM (no feedback on the new API questions though)

File: timesheet_activity_report/__openerp__.py (hunk 1 above)
Comment (seb-elico) [reply timesheet_activity_report/__openerp__.py]: @elicoidal Actually, so far the fields are the same, it's just a port to new API. So I'm wondering if an upgrade of the module is necessary!

Comment (elicoidal) [reply timesheet_activity_report/__openerp__.py]: It depends if you include the change in the SQL view or not in this PR.

Comment (seb-elico) [reply timesheet_activity_report/__openerp__.py]: Indeed I simplified the SQL view (not for the BR yet, I wanted to do that in a separate PR), you're right. Thanks!

**Comment by seb-elico** [comment]
@Elico-Corp/odoo-reviewers Final version, please review

File: timesheet_activity_report/README.rst [hunk 2]
```diff
@@ -15,37 +15,48 @@ Installation
 
 To install this module, you need to:
 
- * have basic modules installed (hr_timesheet, project_timesheet, project_issue_sheet)
-
+#. Have basic modules installed (hr_timesheet, project_timesheet, project_issue_sheet)
```
Comment (elicoidal) [inline timesheet_activity_report/README.rst]: doesnot make sense to make an enumerated list with one item no?

File: timesheet_activity_report/README.rst [hunk 3]
```diff
@@ -15,37 +15,48 @@ Installation
...
 
 Bug Tracker
 ===========
 
-Bugs are tracked on `GitHub Issues <https://github.com/Elico-Corp/odoo/issues>`_.
```
Comment (elicoidal) [inline timesheet_activity_report/README.rst:24]: I miss the runbot

File: timesheet_activity_report/report/timesheet_activity_report.py [hunk 4]
```diff
@@ -2,125 +2,120 @@
...
-    }
+    id = fields.Integer('Id', readonly=True)
+    activity_type = fields.Selection(
+        [
+            ('task', 'Task'),
+            ('issue', 'Issue'),
+            ('timesheet', 'Timesheet'),
+        ], 'Type', readonly=True,
+        help="Type is used to separate Tasks, Issues and "
+             "Timesheets input directly")
```
Comment (elicoidal) [inline timesheet_activity_report/report/timesheet_activity_report.py]: Timesheet direct input

**Review by elicoidal — CHANGES_REQUESTED** [review]
some details. overall LGTM but needs to be tested

File: timesheet_activity_report/README.rst (hunk 2 above)
Comment (seb-elico) [reply timesheet_activity_report/README.rst]: I just copy/pasted the README from another module that also had only 1 item. Any recommendation then?

File: timesheet_activity_report/README.rst (hunk 3 above)
Comment (seb-elico) [reply timesheet_activity_report/README.rst:24]: Project is not under Runbot

File: timesheet_activity_report/report/timesheet_activity_report.py (hunk 4 above)
Comment (seb-elico) [reply timesheet_activity_report/report/timesheet_activity_report.py]: @elicoidal Actually, the full sentence should be:

> "Type is used to separate Timesheets input in a Task, in an Issue or in Timesheet Activities" 

Is that ok for you?

File: timesheet_activity_report/README.rst (hunk 2 above)
Comment (martintorresvm) [reply timesheet_activity_report/README.rst]: IMO this looks cleaner:
```
Have basic modules installed:

   * hr_timesheet
   * project_timesheet
   * project_issue_sheet
```

File: timesheet_activity_report/README.rst (hunk 3 above)
Comment (elicoidal) [reply timesheet_activity_report/README.rst:24]: OK

File: timesheet_activity_report/report/timesheet_activity_report.py (hunk 4 above)
Comment (elicoidal) [reply timesheet_activity_report/report/timesheet_activity_report.py]: Another option (and name could be source or origin maybe but n-t-h)
Type is based on the origin of the input (Task, Issue or Timesheet Activity Report)

**Review by elicoidal — CHANGES_REQUESTED** [review]
some nice to have details

**Comment by seb-elico** [comment]
@victormartinelicocorp Any idea why Travis is failing?

````
File "/home/travis/build/Elico-Corp/odoo-addons/timesheet_activity_report/report/timesheet_activity_report.py", line 121, in init
)""")
File "/home/travis/odoo-8.0/openerp/sql_db.py", line 171, in wrapper
return f(self, *args, **kwargs)
File "/home/travis/odoo-8.0/openerp/sql_db.py", line 247, in execute
res = self._obj.execute(query, params)
ProgrammingError: column p.business_requirement_id does not exist
LINE 58:                         ON b.id = p.business_requirement_id
````

**Comment by seb-elico** [comment]
@victormartinelicocorp I found why it's failing, I just pushed a new `oca_dependencies.txt`. Let's see if it'll fix the bug

**Comment by seb-elico** [comment]
@elicoidal @victormartinelicocorp can you approve please? Let's merge as soon as Travis build is green

**Review by victor-ditay — APPROVED** [review]

**Review by elicoidal — APPROVED** [review]

**Pull request merged**